test/e2e/hardwaretest/hardware_test.go

```diff
@@ -0,0 +1,68 @@
+package hardwaretest
+
+import (
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	"github.com/equinix/metal-cli/internal/hardware"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/spf13/cobra"
+)
+
+func TestCli_Hardware(t *testing.T) {
+	subCommand := "hardware"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  hardware.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				root.SetArgs([]string{subCommand, "get"})
```

Comment:
Looks like we need `-i` or `-p` and an ID here. Since we won't have any hardware reservations available to this E2E account, we should use a mocked client that will always return a stub reservation.


```
=== RUN   TestCli_Hardware
=== RUN   TestCli_Hardware/get
Error: either id or project-id should be set
Usage:
  metal hardware-reservation get [-p <project_id>] | [-i <hardware_reservation_id>] [flags]

Aliases:
  get, list

Examples:
  # Retrieve all hardware reservations of a project:
  metal hardware-reservations get -p $METAL_PROJECT_ID
  
  # Retrieve the details of a specific hardware reservation:
  metal hardware-reservations get -i 8404b73c-d18f-4190-8c49-20bb17501f88

Flags:
  -h, --help                help for get
  -i, --id string           The UUID of a hardware reservation.
  -p, --project-id string   A project's UUID.

Global Flags:
      --config string         Path to JSON or YAML configuration file
      --exclude strings       Comma separated Href references to collapse in results, may be dotted three levels deep
      --filter stringArray    Filter 'get' actions with name value pairs. Filter is not supported by all resources and is implemented as request query parameters.
      --http-header strings   Headers to add to requests (in format key=value)
      --include strings       Comma separated Href references to expand in results, may be dotted three levels deep
  -o, --output string         Output format (*table, json, yaml). env output formats are (*sh, terraform, capp).
      --search string         Search keyword for use in 'get' actions. Search is not supported by all resources.
      --sort-by string        Sort fields for use in 'get' actions. Sort is not supported by all resources.
      --sort-dir string       Sort field direction for use in 'get' actions. Sort is not supported by all resources.
      --token string          Metal API Token (METAL_AUTH_TOKEN)

    hardware_test.go:45: either id or project-id should be set
    hardware_test.go:56: expected output should include n3.xlarge.x86, m3.large.x86, s3.xlarge.x86, dc10.
--- FAIL: TestCli_Hardware (0.00s)
    --- FAIL: TestCli_Hardware/get (0.00s)
FAIL
```

Reply:
The terraform provider has an example of setting up a mock API with `httptest`: https://github.com/equinix/terraform-provider-equinix/blob/main/equinix/resource_metal_device_acc_test.go#L1082-L1116

In this case, you would want to have your mock return a 200 with a JSON string that has the necessary fields.  You could also consider the other mocking approaches documented here: https://medium.com/zus-health/mocking-outbound-http-requests-in-go-youre-probably-doing-it-wrong-60373a38d2aa

Reply:
The test mentioned earlier is still failing with the same error as before:

```
=== RUN   TestCli_Hardware/get
Error: either id or project-id should be set
```